google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITE2ETracingTest.java

```diff
@@ -0,0 +1,549 @@
+  private static Trace retrievedTrace;
+
+  private static String rootSpanName;
+  private static Tracer tracer;
+
+  // Required to set custom-root span
+  private static OpenTelemetrySdk openTelemetrySdk;
+
+  private static String projectId;
+
+  private static DatastoreOptions options;
+
+  private static Datastore datastore;
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    projectId = DatastoreOptions.getDefaultProjectId();
+    logger.info("projectId:" + projectId);
+
+    // TODO(jimit) Make it re-usable w/ InMemorySpanExporter
+    traceExporter =
+        TraceExporter.createWithConfiguration(
+            TraceConfiguration.builder().setProjectId(projectId).build());
+
+    traceClient_v1 = TraceServiceClient.create();
+
+    random = new Random();
+  }
+
+  @Before
+  public void before() throws Exception {
+    // Set up OTel SDK
+    Resource resource =
+        Resource.getDefault().merge(Resource.builder().put(SERVICE_NAME, "Sparky").build());
+
+    if (isUsingGlobalOpenTelemetrySDK()) {
+      openTelemetrySdk =
+          OpenTelemetrySdk.builder()
+              .setTracerProvider(
+                  SdkTracerProvider.builder()
+                      .setResource(resource)
+                      .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                      .setSampler(Sampler.alwaysOn())
+                      .build())
+              .buildAndRegisterGlobal();
+    } else {
+      openTelemetrySdk =
+          OpenTelemetrySdk.builder()
+              .setTracerProvider(
+                  SdkTracerProvider.builder()
+                      .setResource(resource)
+                      .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                      .setSampler(Sampler.alwaysOn())
+                      .build())
+              .build();
+    }
+
+    // Initialize the Datastore DB w/ the OTel SDK. Ideally we'd do this is the @BeforeAll method
+    // but because gRPC traces need to be deterministically force-flushed, datastore.shutdown()
+    // must be called in @After for each test.
+    DatastoreOptions.Builder optionsBuilder;
+    if (isUsingGlobalOpenTelemetrySDK()) {
+      optionsBuilder =
+          DatastoreOptions.newBuilder()
+              .setOpenTelemetryOptions(
+                  DatastoreOpenTelemetryOptions.newBuilder().setTracingEnabled(true).build());
+    } else {
+      optionsBuilder =
+          DatastoreOptions.newBuilder()
+              .setOpenTelemetryOptions(
+                  DatastoreOpenTelemetryOptions.newBuilder()
+                      .setOpenTelemetry(openTelemetrySdk)
+                      .setTracingEnabled(true)
+                      .build());
+    }
+
+    String namedDb = System.getProperty("FIRESTORE_NAMED_DATABASE");
+    if (namedDb != null) {
+      logger.log(Level.INFO, "Integration test using named database " + namedDb);
+      optionsBuilder = optionsBuilder.setDatabaseId(namedDb);
```

Comment:
we should ideally parameterize these tests rather than leave it all in if/else blocks. Please see `ITDatastoreTest` to see how parameterization for multiple databases is set up.

Reply:
These were originally parameterized. But when we rolled these out for Firestore as part of CI/CD, there were some challenges with using parameterization with GraalVM. (IIRC you commented on the issue in chat :)

Reply:
Added back the parameterization.